src/dmd/errors.d

```diff
@@ -401,19 +401,49 @@ extern (C++) void message(const(char)* format, ...)
 }
 
 /**
- * Just print to stderr, doesn't care about gagging.
- * (format,ap) text within backticks gets syntax highlighted.
- * Params:
- *      loc         = location of error
- *      headerColor = color to set `header` output to
- *      header      = title of error message
- *      format      = printf-style format specification
- *      ap          = printf-style variadic arguments
- *      p1          = additional message prefix
- *      p2          = additional message prefix
- */
+* To convert a function pointer into a delegate.
+*/
+auto toDelegate(F)(auto ref F func)
+{
+    static if (is(typeof(*F) Params == __parameters))
+    {
+        return (Params params) {
+            return func(params);
+        };
+    }
+    else
+    {
+        static assert(false, "The given argument `func` is not a function type");
+    }
+}
+
+alias diagnosticHandlerDelegate = void delegate(const ref Loc, Color, const(char)*, const(char)*, va_list, const(char)*, const(char)*) nothrow;
```

Comment:
* The name should be capitalized. That's the standard convention for types
* I don't think `Delegate` needs to be part of the name